### ISSUE TITLE
Postgres sqlc query generation fails for returning properties in the USING table during a DELETE

### ISSUE THREAD

**Issue by buildbreaker** [sqlc-dev/sqlc#1714]
### Version

1.14.0

### What happened?

I'm trying to write a DELETE query which I want to join the results with another table. The way that I am familiar with is to use the `USING` keyword but that seems to be scoped to the `WHERE` clause only. When I try returning the columns from the `USING` table, `sqlc generate` complains that it cant find the column in the `RETURNING` block.

I have an example with CI testing/output [here](https://github.com/buildbreaker/HelloWorld/pull/6). 
[Add sqlc DELETE/USING issue demonstration](https://github.com/buildbreaker/HelloWorld/pull/6/commits/6874f0400325e9bdafb335d2c4a7236edf04c0d8) contains the commit for the "working path". 
[failure case](https://github.com/buildbreaker/HelloWorld/pull/6/commits/709d00786818f4bb078f919f8e515810fe732cb9) introduces the problematic query. The CI error can be found [here](https://github.com/buildbreaker/HelloWorld/runs/7101547962?check_suite_focus=true).

I can get around this specific issue by using CTEs but it'd be nice to get this way working also!

### Relevant log output

```shell
Error message from the "Failing query" in the SQL queries section:

    # package demo
    query.sql:26:15: column "b_id" does not exist
```

### Database schema

```sql
CREATE TABLE a
(
    a_id TEXT,
    b_id_fk TEXT
);
CREATE TABLE b
(
    b_id TEXT
);
```

### SQL queries
Query that works:
```sql
-- name: GetSomeDeletedOk :many
DELETE FROM a
    USING b
WHERE a.b_id_fk = b.b_id
    RETURNING *; -- note no columns from b in model
```

Failing query:
```sql
-- name: GetSomeDeletedNotOk :many
DELETE FROM a
USING b
WHERE a.b_id_fk = b.b_id
RETURNING b.b_id; -- column "b_id" does not exist
```

### Configuration

```yaml
version: 1
packages:
  - path: "demo"
    name: "demo"
    engine: "postgresql"
    schema: "schema.sql"
    queries: "query.sql"
```

### Playground URL

https://play.sqlc.dev/p/1ca4c468637ee1d35988be307ea517680fce1fb63af2b816a5a9af4361bf4f68

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by saquibmian** [comment]
I've been hit by this as well. Are there any workarounds for this?

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/delete_using/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/ce36c2af218a26dcc4246966f9487b88689be6e2bcca1ca20dd48205f2fdbcf8)
